### ISSUE TITLE
command not found: Not supported on Mac?

### ISSUE THREAD

**Issue by simplificate** [MikhailArkhipov/vscode-r#19]
Hi there,

I was kicking the tires on this project and had trouble getting this (and, really, any) R-related extension working on Mac.  I found [this article on RTVS](https://docs.microsoft.com/en-us/visualstudio/rtvs/installing-r-tools-for-visual-studio?view=vs-2017) (from Jan 2018), which said:

> RTVSis not presently supported on Visual Studio for Mac.

I know that this project is derived from RTVS, so I wondered if that was the root cause (although your extension says it supports Mac).  I also tried this on two different Macs and got the same result.  Here are the steps I took:

- Install R (Dec 2018)
- Install .Net Framework (not sure if I need this)
- Install vscode-r extension
- Create new file in Visual Studio Code
- Command Palette, then R: Create R Terminal 

Error: 

> command 'r.createRTerm' not found

I [found this page](https://github.com/Microsoft/vscode/issues/15837) which suggested running an `npm install` in the extension folder.  Here's the path where I found it:

`/.vscode/extensions/mikhail-arkhipov.r-0.0.5`

This didn't help.  Here's a dump of what happened:

```
WARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/isarray-1958e99d/package.json'
WARNWARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/is-glob-c65b67ee/package.json'
WARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/is-extglob-3fd7b62d/package.json'
 tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/isarray-1958e99d/README.md'
WARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/is-glob-c65b67ee/README.md'
WARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/is-extglob-3fd7b62d/README.md'
WARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/isarray-1958e99d/index.js'
WARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/is-glob-c65b67ee/LICENSE'
WARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/is-extglob-3fd7b62d/LICENSE'
WARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/is-glob-c65b67ee/index.js'
WARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/is-extglob-3fd7b62d/index.js'
WARN tar ENOENT: no such file or directory, open '/Users/kenwalke/.vscode/extensions/mikhail-arkhipov.r-0.0.5/node_modules/.staging/typescript-6ee193c8/lib/tsserver.js'
npm ERR! code E404
npm ERR! 404 Not Found: event-stream@3.3.6
```
I have a tech background and have used R Studio, but I'm not an expert in either of these.  I'm an analyst and I mostly use VSC for writing SQL for my day job.  I also understand this is a volunteer project, so I appreciate the help!

Thanks,
Ken

**Comment by Hong-Revo** [comment]
See #7. Waiting a bit before starting the terminal may make the problem go away.

**Comment by MikhailArkhipov** [comment]
Is startup complete as in 

![image](https://user-images.githubusercontent.com/12820357/52538393-e7a70280-2d26-11e9-98ae-553ad0d60177.png)

or the extension is still loading?

**Comment by MikhailArkhipov** [comment]
Also, you may have to save the new file with .r extension first since otherwise VS Code does not know that the file is in R.

**Comment by simplificate** [comment]
@MikhailArkhipov ah, thanks for the pointer to look in the Output for the Startup Log.  Here's what it says:

```
> Checking for .NET Core... OK
> Installing libzip (required for the R process executable)... undefined
```
I'm going to try to install libzip and see if that helps.

**Comment by simplificate** [comment]
Ok, `brew install libzip` appeared to clear the libzip error.  I quit and relaunched VSC and saw this:

**R Tools Startup Log**
```
> Checking for .NET Core... OK
> Activating R Language Server...
```
This time, though, the extension told me with a notification it couldn't be started.  Here's a new error  in the R Tools Output.

**R Tools**

```
> [Error - 5:18:45 PM] Starting client failed
> Launching server using command dotnet failed.
```
Going to google the second line in this log...

**Comment by simplificate** [comment]
I can run `dotnet` from the command line.  Am I missing something?

```
> $ dotnet --list-runtimes
> Microsoft.AspNetCore.All 2.2.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
> Microsoft.AspNetCore.App 2.2.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
> Microsoft.NETCore.App 2.2.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
```

**Comment by MikhailArkhipov** [comment]
Yep, that is missing libzip.

**Comment by simplificate** [comment]
Thank you @MikhailArkhipov. Libzip is successfully installed now, but I can’t make heads or tails of the new error:

```
> [Error - 5:18:45 PM] Starting client failed
> Launching server using command dotnet failed.
```

**Comment by MikhailArkhipov** [comment]
Do you still get `> Segmentation fault: 11`?

**Comment by simplificate** [comment]
@MikhailArkhipov yes, same result

**Comment by hmassalha** [comment]
vscode-insider solved the problem for mac users

**Comment by ECon87** [comment]
Hi, I am facing the same issue. Are there any updates?

**Comment by Hong-Revo** [comment]
Unfortunately it looks like Mikhail has moved on to other projects. This was always meant to be more of a personal project anyway, not a Microsoft-supported extension.

You can try the other R extension for VSCode, which is still actively supported:

https://github.com/Ikuyadeu/vscode-R

It works very well in conjunction with the R LSP client, for tooltip help and language completion features:

https://github.com/REditorSupport/vscode-r-lsp

**Comment by ECon87** [comment]
Thanks for the reply.

I think that I have found a workaround for MacOS. If you start VS Code via the terminal then "R Tools" seems to work (open -a Visual\ Studio\ Code). This is a workaround I found in the "LaTeX workplace" github, and it has to do with the PATH set by the different ways of launching the app.

I get a new error (see below) but r tools seems to work in the editor itself. Maybe I am wrong. 

`[Error - 10:39:41 AM] Connection to server is erroring. Shutting down server.
A fatal error occurred, the folder [/usr/local/share/dotnet/host/fxr] does not contain any version-numbered child folders`
